enclave-runtime/src/lib.rs

```diff
@@ -597,119 +604,186 @@ where
 	O: EnclaveOnChainOCallApi,
 	P: TrustedOperationPool<Hash = H256> + 'static,
 {
 	let author = Author::new(Arc::new(top_pool));
+	let ends_at = duration_now() + max_exec_duration;
 
 	let mut calls = Vec::<OpaqueCall>::new();
-	let mut blocks = Vec::<SignedSidechainBlock>::new();
-	let start_time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs() as i64;
-	let mut is_done = false;
-	for shard in shards.into_iter() {
-		let mut call_hashes = Vec::<H256>::new();
-
-		// load state before executing any calls
-		let mut state = load_initialized_state(&shard)?;
-		// save the state hash before call executions
-		// (needed for block composition)
-		trace!("Getting hash of previous state ..");
-		let prev_state_hash = state::hash_of(state.state.clone())?;
-		trace!("Loaded hash of previous state: {:?}", prev_state_hash);
-
-		// retrieve trusted operations from pool
-		let trusted_calls = author.get_pending_tops_separated(shard)?.0;
-
-		debug!("Got following trusted calls from pool: {:?}", trusted_calls);
-		// call execution
-		for trusted_call_signed in trusted_calls.into_iter() {
-			match handle_trusted_worker_call(
-				&mut calls,
-				&mut state,
-				&trusted_call_signed,
-				latest_onchain_header.clone(),
-				shard,
-				on_chain_ocall,
-			) {
-				Ok(hashes) => {
-					let inblock = match hashes {
-						Some((_, operation_hash)) => {
-							call_hashes.push(operation_hash);
-							true
-						},
-						None => {
-							// remove call as invalid from pool
-							false
-						},
-					};
-
-					// TODO: prune instead of remove_top ? Block needs to be known
-					// TODO: move this pruning to after finalization confirmations, not here!
-					// remove calls from pool (either as valid or invalid)
-					author
-						.remove_top(
-							vec![TrustedOperationOrHash::Operation(
-								trusted_call_signed.into_trusted_operation(true),
-							)],
-							shard,
-							inblock,
-						)
-						.unwrap();
-				},
-				Err(e) =>
-					error!("Error performing worker call (will not push top hash): Error: {:?}", e),
-			};
-			// Check time
-			if time_is_overdue(Timeout::Call, start_time) {
-				is_done = true;
-				break
-			}
-		}
-		// create new block (side-chain)
-		match compose_block_and_confirmation(
-			latest_onchain_header.clone(),
-			call_hashes,
-			shard,
-			prev_state_hash,
+	let mut call_hashes = Vec::<H256>::new();
+
+	// load state before executing any calls
+	let mut state = load_initialized_state(&shard)?;
+	// save the state hash before call executions
+	// (needed for block composition)
+	trace!("Getting hash of previous state ..");
+	let prev_state_hash = state::hash_of(state.state.clone())?;
+	trace!("Loaded hash of previous state: {:?}", prev_state_hash);
+
+	// retrieve trusted operations from pool
+	let trusted_calls = author.get_pending_tops_separated(shard)?.0;
+
+	debug!("Got following trusted calls from pool: {:?}", trusted_calls);
+	// call execution
+	for tcs in trusted_calls.into_iter() {
```

Comment:
I'm not a huge fan of abbreviations like `tcs` here. A reader of the code then has to somehow deduce what it means and keep it in mind while reading the rest of the loop. I'd prefer the unambiguous `trusted_call_signed`.

Reply:
O kind of agree. I think it is a trade-off because long names also clutter the view, and make it tiresome to read. But yes, I think I will change it to the long name.

Reply:
Changed that. I remember why I did this change; It made a longer statement fit on one line. 😃 

But it's better your way.

Reply:
Haha, laughed out loud. 😄